src/platform/react-hooks/src/hooks/usePresenceEnter.ts

```diff
@@ -0,0 +1,59 @@
+import type * as Ably from 'ably';
+import { useCallback, useEffect } from 'react';
+import { ChannelParameters } from '../AblyReactHooks.js';
+import { useAbly } from './useAbly.js';
+import { useChannelInstance } from './useChannelInstance.js';
+import { useStateErrors } from './useStateErrors.js';
+
+export interface PresenceEnterResult<T> {
+  updateStatus: (messageOrPresenceObject: T) => void;
+  connectionError: Ably.ErrorInfo | null;
+  channelError: Ably.ErrorInfo | null;
+}
+
+const INACTIVE_CONNECTION_STATES: Ably.ConnectionState[] = ['suspended', 'closing', 'closed', 'failed'];
+
+export function usePresenceEnter<T = any>(
```

Comment:
not sure about naming for new hooks, open to suggestions:
`usePresenceEnter` - for `presence.enter()` logic
`usePresenceListener` - for `presence.subscribe(['enter', 'leave', 'update']` logic

Reply:
Decided to rename `usePresenceEnter` to `usePresence`. Clashing with the previous meaning of `usePresence` is justifiable since we're releasing those changes in a major version
`usePresenceListener` is fine.